### PR TITLE
HPCC-27539 Ensure defaultCopies in containerized mode = 1

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please comment/review.

File: dali/base/dafdesc.cpp [hunk 1]
```diff
@@ -101,7 +101,9 @@ void ClusterPartDiskMapSpec::setRoxie (unsigned redundancy, unsigned channelsPer
 {
     flags = 0;
     replicateOffset = _replicateOffset?_replicateOffset:1;
-    defaultCopies = redundancy+1;
+#ifdef _CONTAINERIZED
```
Comment (ghalliday) [inline dali/base/dafdesc.cpp:104]: question: Should this still be setting defaultCopies?

File: dali/base/dafdesc.cpp [hunk 2]
```diff
@@ -223,6 +225,9 @@ void ClusterPartDiskMapSpec::fromProp(IPropertyTree *tree)
     }
     replicateOffset = getPropDef(tree,"@replicateOffset",1);
     defaultCopies = getPropDef(tree,"@redundancy",defrep)+1;
+#ifdef _CONTAINERIZED
+    defaultCopies = 1;
```
Comment (ghalliday) [inline dali/base/dafdesc.cpp:229]: The #ifdefs could do with a comment indicating why this is being done.

**Review by ghalliday — COMMENTED** [review]
@jakesmith

File: dali/base/dafdesc.cpp (hunk 1 above)
Comment (jakesmith) [reply dali/base/dafdesc.cpp:104]: The defaultCopies default value in containerized mode is now 1, so this change code just leaves it as that.

File: dali/base/dafdesc.cpp (hunk 2 above)
Comment (jakesmith) [reply dali/base/dafdesc.cpp:232]: @ghalliday - have added some comments

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash

**Comment by jakesmith** [comment]
Squashed
@richardkchapman - ready to merge.

**Pull request merged**